File: src/main/java/dev/lbuddyboy/flash/rank/comparator/RankWeightComparator.java
package dev.lbuddyboy.flash.rank.comparator;

import dev.lbuddyboy.flash.rank.Rank;

import java.util.Comparator;

public class RankWeightComparator implements Comparator<Rank> {

    @Override
    public int compare(Rank rank, Rank otherRank) {
        return Integer.compare(rank.getWeight(), otherRank.getWeight());
    }
}

File: src/main/java/dev/lbuddyboy/flash/rank/comparator/RankWeightComparator.kt
package dev.lbuddyboy.flash.rank.comparator

import dev.lbuddyboy.flash.rank.Rank

class RankWeightComparator : Comparator<Rank?> {
    override fun compare(rank: Rank?, otherRank: Rank?): Int {
        return Integer.compare(rank.getWeight(), otherRank.getWeight())
    }
}